src/test/scala/mesosphere/marathon/storage/migration/MigrationTest.scala

```diff
@@ -48,35 +49,31 @@ class MigrationTest extends AkkaUnitTest with Mockito with GivenWhenThen with Ev
     val clock = SettableClock.ofNow()
     implicit val scheduler = new SimulatedScheduler(clock)
     val notificationCounter = new AtomicInteger(0)
+    val migrationSteps = if (fakeMigrations.nonEmpty) fakeMigrations else Migration.steps
     val migration = new Migration(Set.empty, None, "bridge-name", persistenceStore, appRepository, podRepository, groupRepository, deploymentRepository,
       instanceRepository, taskFailureRepository, frameworkIdRepository,
-      serviceDefinitionRepository, configurationRepository, backup, config) {
+      serviceDefinitionRepository, configurationRepository, backup, config, migrationSteps) {
 
       override protected def notifyMigrationInProgress(from: StorageVersion, migrateVersion: StorageVersion): Unit =
         notificationCounter.incrementAndGet()
 
-      override def migrations: List[(StorageVersion, () => Future[Any])] = if (fakeMigrations.nonEmpty) {
-        fakeMigrations
-      } else {
-        super.migrations
-      }
     }
   }
 
-  val currentVersion: StorageVersion = StorageVersions.current
+  val currentVersion: StorageVersion = StorageVersions(Migration.steps)
 
   "Migration" should {
     "be filterable by version" in {
       val f = new Fixture
 
       val migrate = f.migration
-      val all = migrate.migrations.filter(_._1 > StorageVersions(0, 0, 0)).sortBy(_._1)
-      all should have size migrate.migrations.size.toLong
+      val all = migrate.steps.filter(_._1 > StorageVersions(0, 0, 0)).sortBy(_._1)
```

Comment:
since we are using _1 on so many places for this, maybe we can make it a case class?

Reply:
Done